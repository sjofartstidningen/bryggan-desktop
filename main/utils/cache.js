'use strict';

const windows = new Map();

module.exports = { windows };
