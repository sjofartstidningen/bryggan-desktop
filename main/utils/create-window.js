'use strict';

const { BrowserWindow } = require('electron');
const ipc = require('electron-better-ipc');
const { windows } = require('./cache');
const { loadRoute } = require('./routes');

const defaultWindowOptions = {
  show: false,
};

function createWindow(name, windowOptions) {
  const isCreated = () => windows.has(name);
  const getWindow = () => windows.get(name);

  const create = () => {
    const win = new BrowserWindow(
      Object.assign({}, defaultWindowOptions, windowOptions),
    );

    windows.set(name, win);
    loadRoute(win, name);

    ipc.answerRenderer(`${name}-ready`, () => win.show());
  };

  const close = () => {
    if (isCreated()) {
      const win = getWindow();
      win.destroy();
      windows.delete(name);
    }
  };

  const hide = () => {
    if (isCreated()) {
      const win = getWindow();
      win.hide();
    }
  };

  const show = () => {
    if (isCreated()) {
      const win = getWindow();
      win.show();
    }
  };

  const toggle = () => {
    if (isCreated()) {
      close();
    } else {
      create();
    }
  };

  const toggleHidden = () => {
    if (!isCreated()) {
      create();
    } else {
      const win = getWindow();

      if (win.isVisible()) {
        hide();
      } else {
        show();
      }
    }
  };

  return {
    isCreated,
    getWindow,
    create,
    close,
    hide,
    show,
    toggle,
    toggleHidden,
  };
}

module.exports = { createWindow };
