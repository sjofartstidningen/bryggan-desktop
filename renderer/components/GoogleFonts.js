import React from 'react';

function GoogleFonts({ fonts }) {
  const fontsFragment = fonts
    .map(font => {
      let name = font.name.replace(/\s/g, '+');
      if (font.weights) {
        name += `:${font.weights.join(',')}`;
      }

      return name;
    })
    .join('|');

  const href = `https://fonts.googleapis.com/css?family=${fontsFragment}`;

  return <link key="google-fonts" rel="stylesheet" href={href} />;
}

export { GoogleFonts };
