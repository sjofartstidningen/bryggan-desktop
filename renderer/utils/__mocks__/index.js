const createSimpleCache = () => {
  const cache = new Map();

  return {
    set: jest.fn((key, value) => cache.set(key, value)),
    get: jest.fn(key => cache.get(key)),
    has: jest.fn(key => cache.has(key)),
    delete: jest.fn(key => cache.delete(key)),
    clear: jest.fn(() => cache.clear()),
  };
};

module.exports = { createSimpleCache };
