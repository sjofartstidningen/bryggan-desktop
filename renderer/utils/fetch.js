const fetch = window && window.fetch ? window.fetch : require('node-fetch');

const AbortController =
  window && window.AbortController
    ? window.AbortController
    : require('abort-controller');

export { fetch, AbortController };
