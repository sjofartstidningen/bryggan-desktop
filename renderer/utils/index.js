import { extname } from 'path';

const sortByType = (a, b) => {
  if (a.type !== b.type) return a.type > b.type ? -1 : 1;

  const aExt = extname(a.name);
  const bExt = extname(b.name);
  if (aExt !== bExt) return aExt > bExt ? 1 : -1;
  if (a.name > b.name) return 1;
  if (a.name < b.name) return -1;
  return 0;
};

export { sortByType };
