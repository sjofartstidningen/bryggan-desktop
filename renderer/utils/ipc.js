import log from 'electron-log';

const callMain = async (...args) => {
  const ipc = require('electron-better-ipc');
  log.verbose(`Call main on channel: ${args[0]}`);
  return ipc.callMain(...args);
};

const answerMain = (...args) => {
  const ipc = require('electron-better-ipc');
  log.verbose(`Setup answer to main on channel: ${args[0]}`);
  return ipc.answerMain(...args);
};

export { callMain, answerMain };
