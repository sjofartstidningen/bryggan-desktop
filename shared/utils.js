import path from 'path';

const hasIdlkFile = (file, items) => {
  /**
   * .idlk files are special in that the are somewhat similar to the original
   * filename â€“ but not exactly
   * The original filename is capped at 18 chars and transformed to lower case
   * then wrapped in tildes: "~[original_filename]~[random_chars].idlk"
   */
  const basename = path
    .basename(file.name, '.indd')
    .slice(0, 18)
    .toLowerCase();

  for (const item of items) {
    if (path.extname(item.name) === '.idlk') {
      const [, strippedIdlkName] = item.name.match(/~(.+)~/);
      if (strippedIdlkName === basename) return true;
    }
  }

  return false;
};

export { hasIdlkFile };
