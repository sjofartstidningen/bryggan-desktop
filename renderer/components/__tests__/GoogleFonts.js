import React from 'react';
import { renderAndMatch } from '../../../tests/utils';
import { GoogleFonts } from '../GoogleFonts';

const fonts = [{ name: 'Roboto', weights: [500, 700] }, 'Roboto Mono'];
renderAndMatch('GoogleFonts', <GoogleFonts fonts={fonts} />);
