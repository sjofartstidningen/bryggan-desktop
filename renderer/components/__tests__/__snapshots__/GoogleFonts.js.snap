// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GoogleFonts />: should render without issues 1`] = `
<link
  href="https://fonts.googleapis.com/css?family=Roboto:500,700|Roboto+Mono"
  rel="stylesheet"
/>
`;
